framework/src/executioners/Transient.C

```diff
@@ -386,10 +386,10 @@ Transient::execute()
   if (!_app.halfTransient())
   {
     TIME_SECTION(_final_timer);
-
+    _problem.execMultiApps(EXEC_FINAL);
     _problem.finalizeMultiApps();
-    _problem.outputStep(EXEC_FINAL);
     _problem.execute(EXEC_FINAL);
+    _problem.outputStep(EXEC_FINAL);
```

Comment:
Any particular reason for the switch around here? Seems arbitrary, I'm not sure if the new or the old way is any better, more correct, etc.

Reply:
Yes, @joshuahansel will be adding a test. Currently, stuff running on final will run after the final output.

Reply:
oh, of course...